src/main/java/ch/uzh/ifi/hase/soprafs24/controller/UserController.java

```diff
@@ -44,4 +44,25 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
     // convert internal representation of user back to API
     return DTOMapper.INSTANCE.convertEntityToUserGetDTO(createdUser);
   }
+
+  @PutMapping("/users/{id}")
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public UserGetDTO updateUser(@PathVariable Long id, @RequestBody UserPostDTO userPostDTO) {
+    // convert API user to internal representation
```

Comment:
add auth check if auth token is user's token

Reply:
This has been modified now using the code line below, can you check if it's okay?

```
// Check if the token is valid and belongs to the user
    authorizationService.isAuthorized(token);
```

Reply:
this now only checks if the token is in the db. we should also check if the token is the user's token. use this instead:
```java
authorizationService.isAuthorized(token, id);`
```

Reply:
ok